test/indexing_offset.jl

```diff
@@ -23,7 +23,12 @@ using Test, DataFrames, OffsetArrays
     @test_throws DimensionMismatch df[!, :b] .= ov1
 
     # this is consequence of the fact that OffsetArrays wrap AbstractRange in this case
-    @test_throws ErrorException df[:, :a] = ov1
+    # Base.CanonicalIndexError is not available in Julia 1.7 or earlier
+    if isdefined(Base, :CanonicalIndexError)
```

Comment:
I tend to prefer `VERSION` checks as they can potentially be removed automatically when dropping 1.7 support (in 2030? :-).
```suggestion
    if VERSION >= 1.8-DEV
```